XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+
+		[Export ("controllerWillCheckoutViaWeb:")]
+		void ControllerWillCheckoutViaWeb (ViewController viewController);
+
+		[Export ("controllerWillCheckoutViaApplePay:")]
+		void ControllerWillCheckoutViaApplePay (ViewController viewController);
+
+	}
+
+	[BaseType (typeof(UIViewController), Name = "BUYViewController")]
+	[DisableDefaultCtor]
+	interface ViewController : IPKPaymentAuthorizationViewControllerDelegate
+	{
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		IViewControllerDelegate Delegate { get; set; }
+
+		[Export ("client", ArgumentSemantic.Strong)]
+		BuyClient Client { get; set; }
+
+		[Export ("shop", ArgumentSemantic.Strong)]
+		Shop Shop { get; set; }
+
+		[Export ("merchantId", ArgumentSemantic.Strong)]
+		string MerchantId { get; set; }
+
+		[Export ("isApplePayAvailable")]
+		bool IsApplePayAvailable { get; }
+
+		[Export ("allowApplePaySetup")]
+		bool AllowApplePaySetup { get; set; }
+
+		[Export ("canShowApplePaySetup")]
+		bool CanShowApplePaySetup { get; }
+
+		[Export ("shouldShowApplePayButton")]
+		bool ShouldShowApplePayButton { get; }
+
+		[Export ("shouldShowApplePaySetup")]
+		bool ShouldShowApplePaySetup { get; }
+
+		[Export ("checkout", ArgumentSemantic.Strong)]
+		Checkout Checkout { get; }
+
+		[Export ("loadShopWithCallback:"), Async]
+		void LoadShop (Action<bool, NSError> block);
+
+		[Export ("supportedNetworks", ArgumentSemantic.Copy)]
+		NSString[] SupportedNetworks { get; set; }
+
+		[Export ("paymentRequest")]
+		PKPaymentRequest PaymentRequest { get; }
+
+		[Export ("initWithClient:")]
+		IntPtr Constructor (BuyClient client);
+
+		[Export ("startApplePayCheckout:")]
+		void StartApplePayCheckout (Checkout checkout);
+
+		[Export ("startWebCheckout:")]
+		void StartWebCheckout (Checkout checkout);
+
+		[Export ("startCheckoutWithCartToken:")]
+		void StartCheckout (string token);
+
+		[Export ("checkoutCompleted:status:")]
+		void CheckoutCompleted (Checkout checkout, Status status);
+
+		[Static]
+		[Export ("completeCheckoutFromLaunchURL:")]
+		void CompleteCheckout (NSUrl url);
+
+	}
+
+	[BaseType (typeof(ViewController), Name = "BUYProductViewController")]
+	interface ProductViewController : Themeable
+	{
+		[Export ("initWithClient:theme:")]
+		IntPtr Constructor (BuyClient client, Theme theme);
+
+		[Export ("loadProduct:completion:"), Async]
+		void LoadProduct (string productId, Action<bool, NSError> completion);
+
+		[Export ("loadWithProduct:completion:"), Async]
+		void LoadProduct (Product product, Action<bool, NSError> completion);
+
+		[Export ("productId", ArgumentSemantic.Strong)]
+		string ProductId { get; }
+
+		[Export ("product", ArgumentSemantic.Strong)]
+		Product Product { get; }
+
+		[Export ("isLoading")]
+		bool IsLoading { get; }
+
+		[Export ("presentPortraitInViewController:")]
+		void PresentPortraitInViewController (UIViewController controller);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewErrorView")]
+	interface ProductViewErrorView
```

Comment:
Checking with header, I would suggest to add [DisableDefaultCtor] attribute because it would make it easier to configure the theme using Constructor (Theme theme).